### DISCUSSION TITLE
Github Actions randomly skips "This check was skipped"

### DISCUSSION THREAD

**Discussion by smithw-adsk** [community/community#26019] [Actions]
<p>Sometimes an action that is triggered by assignment will not run and the message is “This check was skipped”. Usually we simply un-assign and re-assign and the action launches. We see this regularly.</p>
<p>You can see examples here (PR 717): <a href="https://github.com/Autodesk/maya-usd/actions" rel="nofollow noopener">https://github.com/Autodesk/maya-usd/actions</a><br>
The first two fail and the third launches. There’s no log. Where can I check what happened?</p>
<p>There is a trigger and a check.</p>
<p>Trigger<br>
on:<br>
pull_request:<br>
branches: [dev]<br>
types:    [assigned]</p>
<p>Check<br>
if: contains(toJson(github.event.pull_request.assignee), ‘ecp-maya-devops-adsk’)</p>
<p>Thanks,<br>
William</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/smithw-adsk">@smithw-adsk</a>,</p>
<p>According to the <strong>if</strong> conditional you set in your workflow,</p>
<pre><code class="lang-yaml">if: contains(toJson(github.event.pull_request.assignee), 'ecp-maya-devops-adsk')
</code></pre>
<p>it will check the whole context of the “<strong>github.event.pull_request.assignee</strong>” object to see if it contains the string or sub-string “<strong>ecp-maya-devops-adsk</strong>”.<br>
The “<strong>github.event.pull_request.assignee</strong>” object generally contains the information of the earliest added assignee existing in current <strong>Assignees</strong>.</p>
<p>What is your real purpose?</p>
<ol>
<li>
<p>You want the job (check) in the workflow only runs when the latest added assignee is the user named “<strong>ecp-maya-devops-adsk</strong>”.<br>
If so, the expression you are using is not accurate and strict. You can use the expression “<strong>github.event.assignee.login</strong>” to get the login name of the latest added assignee, and the <strong>if</strong> conditional should look like this.</p>
<pre><code class="lang-yaml">if: ${{ github.event.assignee.login == 'ecp-maya-devops-adsk' }}
</code></pre>
</li>
<li>
<p>You want the job (check) in the workflow only runs when the <strong>Assignees</strong> contains the user named “ecp-maya-devops-adsk”.<br>
If so, you can use the expression “<strong>github.event.pull_request.assignees</strong>” to get the <strong>assignees</strong> object which is an array contains the information of all the existing assignees. And use the expression “<strong>github.event.pull_request.assignees.*.login</strong>” to extract the login names of the assignees into an array from the <strong>assignees</strong> object.<br>
Then, the <strong>if</strong> conditional should look like this.</p>
<pre><code class="lang-yaml">if: ${{ contains(github.event.pull_request.assignees.*.login, 'ecp-maya-devops-adsk') }}
</code></pre>
</li>
</ol>

**Comment by smithw-adsk** [comment]
<p><a class="mention" href="/brightran">@brightran</a></p>
<p>Your suggestions are more accurate and our requirement is the first one you listed. It’s been very reliable in testing and I’ll put it into our action.</p>
<p>Thanks for your help.<br>
William</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/smithw-adsk">@smithw-adsk</a>,</p>
<p>So, this <strong>if</strong> conditional can solve the problem you are facing and meet your requests.</p>
<pre><code class="lang-yaml">if: ${{ github.event.assignee.login == 'ecp-maya-devops-adsk' }}
</code></pre>
<p>Glad that my suggestions can help you.</p>
<p>If possible, you can mark my above suggestions as solution of this topic, so that other users who have the similar problems can notice these suggestions. Thanks.</p>

**Comment by smithw-adsk** [comment]
<p>Hello brightran,</p>
<p>Your suggestion seems to have solved the issue for us.</p>
<pre><code class="lang-auto">if: ${{ github.event.assignee.login == 'ecp-maya-devops-adsk' }}
</code></pre>
<p>Thanks for your help!<br>
William</p>
